src/04-meet-your-hardware/README.md

```diff
@@ -8,40 +8,79 @@ Let's get familiar with the hardware we'll be working with.
-  - A single core ARM Cortex-M4F processor with hardware support for single precision floating point
-    operations and a maximum clock frequency of 72 MHz.
-
-  - 256 KiB of "Flash" memory. (1 KiB = 10**24** bytes)
-
-  - 48 KiB of RAM.
-
-  - many "peripherals": timers, GPIO, I2C, SPI, USART, etc.
-
-  - lots of "pins" that are exposed in the two lateral "headers".
-
-  - **IMPORTANT** This microcontroller operates at (around) 3.3V.
-
-- An [accelerometer] and a [magnetometer][] (in a single package).
+We'll refer to this board as "F3" throughout this book. Here are some of the
+many components on the board:
+
+- A number of LEDs, including the eight aligned in a "compass" formation.
+- Two buttons.
+- Two USB ports.
+- An [accelerometer].
+- A [magnetometer].
+- A [gyroscope].
 
 [accelerometer]: https://en.wikipedia.org/wiki/Accelerometer
 [magnetometer]: https://en.wikipedia.org/wiki/Magnetometer
-
-- A [gyroscope].
-
 [gyroscope]: https://en.wikipedia.org/wiki/Gyroscope
 
-- 8 user LEDs arranged in the shape of a compass
-
-- A second microcontroller: a STM32F103CBT. This microcontroller is actually part of an on-board
-  programmer and debugger named ST-LINK and is connected to the USB port named "USB ST-LINK".
-
-- There's a second USB port, labeled "USB USER" that is connected to the main microcontroller, the
-  STM32F303VCT6, and can be used in applications.
+Of these components, the most important is the [microcontroller] (sometimes
+shortened to "MCU" for "microcontroller unit"), which is the large black square
+sitting in the center of your board. The MCU is what runs your code. You might
+sometimes read about "programming a board", when in reality what we are doing
+is programming the MCU that is installed on the board.
+
+[microcontroller]: https://en.wikipedia.org/wiki/Microcontroller
+
+## STM32F303VCT6 (the "STM32F3")
+
+Since the MCU is so important, let's take a closer look at the one sitting on our board.
+
+Our MCU is surrounded by 100 tiny metal **pins**. These pins are connected to
+**tracings**, the little "roads" that act as the wires connecting components
+together on the board. The MCU can dynamically alter the electrical properties
+of the pins. This works similar to a lightswitch altering how electrical
+current flows through a circuit. By enabling or disabling electrical current to
+flow through a specific pin, an LED attached to that pin (via the tracing) can
+be turned on and off.
+
+By looking at the part number and doing some research, we can learn more
+information about electronic components. The part number of our microcontroller
+starts with `STM32`, which indicates that it was manufactured by [ST
+Microelectronics]. It also lets us know that this is an Arm®-based 32-bit
+microcontroller. The `F3` in the part number indicates that the microcontroller
+is based on the Cortex®-M4. The rest of the part number provides more specifics
+(e.g., extra features, RAM size), and aren't as important to us right now.
```

Comment:
Do you think it's worth linking to the [product page](https://www.st.com/en/microcontrollers-microprocessors/stm32f303vc.html) for this MCU?

Reply:
We may also want to note that this naming scheme (and the information you pulled from it), varies between manufacturers, which each have their own naming conventions.

Reply:
@jamesmunns That seems like good info to have. I tried to tweak the language to include it in https://github.com/rust-embedded/discovery/pull/230/commits/042dfbb28bc4b970d5448af6ef6f5299ff363ee1.

@adamgreig I wouldn't be against it. My only concern is that it's a URL to the site probably controlled by their marketing department, which (in my experience, at least) is more likely to  be left broken at some point than, say, the URL to a datasheet behind a "/resources" path.